keps/sig-node/20200203-memory-manager.md

```diff
@@ -0,0 +1,746 @@
+
+#### Pre-Reserved Memory Zone Flag
+
+This will be also followed by a Kubelet Flag for the Memory Manager configuration, which is described above in [The configuration of Memory Manager](#the-configuration-of-memory-manager) section.
+
+`--pre-reserved-memory-zone=[{numa-node=int, memory-type=string, limit=string}][,][...]`
+
+Example:
+   `--pre-reserved-memory-zone= {numa-node=0, memory-type=memory, limit=1Gi}, {numa-node=1, memory-type=memory, limit=1Gi}`
+
+#### Multi-NUMA Groups Flag
+`--multi-numa-groups` - to configure disjoint groups of NUMA nodes, used for preferred multi-NUMA node memory allocation
+
+Example:
+`--multi-numa-groups=[[0, 4], [1, 5]]`
+
+<!--
+By design, *Multi-NUMA* strategy must treat a group (a group of NUMA nodes) as a unit. *Multi-NUMA* strategy will allocate the memory from more than one NUMA node if needed. 
+-->
+
+The groups must be disjoint, however, as shown in the figure in [this](#design-overview) section. For instance, `[0]`, `[1, 2]`, `[3]` groups are disjoint. This is because the groups that overlap do not ensure guaranteed memory allocation across multiple NUMA nodes (e.g. `[0, 1]`, `[1, 2]`, `[3]`). In other words, if a pair of groups overlap, memory resources of some group can be prior consumed by another group so that the other group will starve.
+
+### Test Plan
+The Memory Manager E2E test will enable the Topology Manager feature gate and set the Memory Manager policy to 'static'.
+
+At the beginning of the test, the code will determine if the system under test has support for single or multi-NUMA nodes.
+
+Both Single-NUMA and Multi-NUMA tests will:
+- be carried out for each of the four Topology Manager policies
+- be spinning up non-guaranteed (BestEffort, Burstable) pods, guaranteed pods, and multiple guaranteed and non-guaranteed pods
+- be carried out for all policies and all QoS classes, tests will validate whether pods can be deployed without errors
+
+#### Single-NUMA Systems Tests
+The admission of pods will be validated. Tests related to multi-NUMA systems will be skipped. 
+
+#### Multi-NUMA System Tests
+Memory pinning will be validated for Topology Manager `single-numa-node` and `restricted` policies. Once a pod in Guaranteed/QoS class is deployed, the NUMA node affinity for CPU assignment, memory, hugepages, or devices will be validated. Specifically for Topology Manager `restricted` policy, multi-NUMA feature will be tested against memory demands exceeding the capacity of a single NUMA node. 
+
+### Graduation Criteria
+
+#### Phase 1: Alpha (v1.19)
+- Feature gate is disabled by default.
+- Memory Manager supports multi-NUMA guaranteed memory allocation (incorporates single-numa allocation) for pods in Guaranteed/QoS class.
+- Unit test coverage.
+- Memory Manager allocation policy takes topology hints into account.
+- Alpha-level documentation.
+- All flags are supported, i.e.: `--memory-manager-policy`, `--pre-reserved-memory-zone`, `--multi-numa-groups`.
```

Comment:
nope it should not be supported, because we do not have an agreement on what to do with default value, so we decided not to add it for now and leave it for beta once we will have a proper solution

Reply:
Done. I removed it from the KEP because, in fact, the conclusion was towards removing the flag.
I mean, the conclusion above: https://github.com/kubernetes/enhancements/pull/1203#issuecomment-661792150 (it contains the reasoning why the flag should be removed based on the opinions from different people)
The flag should be removed in favor of NUMA distances feature (a further incremental change).
@cynepco3hahue as far as I remember we agreed on Slack towards removing the flag :)